kubernetes-ingress.el

```diff
@@ -0,0 +1,136 @@
+;;; kubernetes-ingress.el --- Rendering routines for Kubernetes ingress.  -*- lexical-binding: t; -*-
+;;; Commentary:
+;;; Code:
+
+(require 'dash)
+
+(require 'kubernetes-ast)
+(require 'kubernetes-loading-container)
+(require 'kubernetes-modes)
+(require 'kubernetes-props)
+(require 'kubernetes-state)
+(require 'kubernetes-utils)
+(require 'kubernetes-yaml)
+
+
+;; Components
+
+(defconst kubernetes-ingress--column-heading
+  (propertize (format "%-45s %6s" "Name" "Age")
+              'face 'magit-section-heading))
+
+(kubernetes-ast-define-component ingress-detail (ingress)
+  (-let [(&alist 'metadata (&alist 'namespace ns 'creationTimestamp time)) ingress]
+    `((section (namespace nil)
+               (nav-prop (:namespace-name ,ns)
+                         (key-value 12 "Namespace" ,(propertize ns 'face 'kubernetes-namespace))))
+      (key-value 12 "Created" ,time))))
+
+(kubernetes-ast-define-component ingress-line (state ingress)
+  (-let* ((current-time (kubernetes-state-current-time state))
+          (pending-deletion (kubernetes-state-ingress-pending-deletion state))
+          (marked-ingress (kubernetes-state-marked-ingress state))
+          ((&alist 'data data 'metadata (&alist 'name name 'creationTimestamp created-time))
+           ingress)
+          (line `(line ,(concat
+                         ;; Name
+                         (format "%-45s " (kubernetes-utils-ellipsize name 45))
+
+                         ;; Age
+                         (let ((start (apply #'encode-time (kubernetes-utils-parse-utc-timestamp created-time))))
+                           (propertize (format "%6s" (kubernetes-utils-time-diff-string start current-time))
+                                       'face 'magit-dimmed))))))
+
+    `(nav-prop (:ingress-name ,name)
+               (copy-prop ,name
+                          ,(cond
+                             ((member name pending-deletion)
+                              `(propertize (face kubernetes-pending-deletion) ,line))
+                             ((member name marked-ingress)
+                              `(mark-for-delete ,line))
+                            (t
+                             line))))))
+
+(kubernetes-ast-define-component ingress (state ingress)
+  `(section (,(intern (kubernetes-state-resource-name ingress)) t)
+            (heading (ingress-line ,state ,ingress))
+            (section (details nil)
+                     (indent
+                      (ingress-detail ,ingress)
+                      (padding)))))
+(kubernetes-ast-define-component ingress-list (state &optional hidden)
+  (-let [(&alist 'items ingress) (kubernetes-state-ingress state)]
+    `(section (ingress-container ,hidden)
+              (header-with-count "ingress" ,ingress)
```

Comment:
nit, All the header starts with  capital letters. I think if we use "Ingress" we can be consistent.

Reply:
Thanks. Updated it.